lib/game.rb

```diff
@@ -0,0 +1,97 @@
+require 'pry'
+
+class Game
+  attr_accessor :board, :player_1, :player_2, :current_player
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+  def initialize( player_1 = Human.new("X"), player_2 = Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    turns = @board.turn_count
+    # binding.pry
+    if turns % 2 == 0
+      @current_player = @player_1
+    else 
+      @current_player = @player_2
+    end
+    return @current_player
+  end
+
+  def over?
+    if draw? || @board.full? || won? 
+      return true
+    else 
+      return false
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |combo|
+      position_1 = @board.cells[combo[0]]
+      position_2 = @board.cells[combo[1]]
+      position_3 = @board.cells[combo[2]]
+
+      if(position_1 == "X" && position_2 == "X" && position_3 == "X" || position_1 == "O" && position_2 == "O" && position_3 == "O")
+        puts "Congratulations #{position_1}!"
+        return combo
+      end
+    end
+    return false
```

Comment:
You should never `return` from within a loop. If you need a return value from within a loop you need to use a higher level iterator or some sort of local variable. 

Consider this solution to a simpler problem of finding the even number in an array.

```ruby
the_even_number = nil
[1,2,3].each do |i|
  if i.even?
    the_even_number = i
  end
end
the_even_number
```

Instead of returning from within that loop, we flip the switch local variable `the_even_number` to the element in the array that made the if statement true. Then we leave `the_even_number` as the last line to ensure the return value. If no even numbers are present, `the_even_number` will remain nil.

But better, let's just use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect)

```ruby
the_even_number = [1,2,3].detect do |i|
  i.even?
end
```